Microchip/Winc15x0/GHIElectronics.TinyCLR.Drivers.Microchip.Winc15x0/Winc15x0Interface.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Runtime.CompilerServices;
+using GHIElectronics.TinyCLR.Devices.Network;
+
+namespace GHIElectronics.TinyCLR.Drivers.Microchip.Winc15x0 {
+    public class Winc15x0Interface {
+        public NetworkController Controller { get; }
+
+        public Winc15x0Interface() => this.Controller = NetworkController.FromName("GHIElectronics.TinyCLR.NativeApis.ATWINC15xx.NetworkController");
+
+        ~Winc15x0Interface() => this.Dispose();
+
+        public void Dispose() {
+            this.Controller.Dispose();
+            GC.SuppressFinalize(this);
+        }
+
+        public string GetFirmwareVersion() {
+            this.NativeReadFirmwareVersion(out var ver1, out var ver2);
+
+            var major = (ver1 >> 16) & 0xFF;
+            var monor = (ver1 >> 8) & 0xFF;
+            var path = (ver1 >> 0) & 0xFF;
+
+            return major.ToString() + "." + monor.ToString() + "." + path.ToString() + " Svnrev " + ver2.ToString();
+        }
+
+        public bool TurnOn() => this.NativeTurnOn();
+        public uint GetFlashSize() => this.NativeGetFlashSize();
+        public uint ReadChipId() => this.NativeReadChipId();
+        public void ReadFirmwareVersion(out uint ver1, out uint ver2) => this.ReadFirmwareVersion(out ver1, out ver2);
+        public bool FirmwareUpdatebyOta(string url, int timeout) => this.NativeFirmwareUpdatebyOta(url, timeout);
+        public bool FirmwareUpdate(byte[] data, int offset, int count) {
```

Comment:
Add a newline before this method and add an overload that doesn't take offset or count and just forwards here.